include/signalrclient/hub_connection.h

```diff
@@ -46,6 +50,12 @@ namespace signalr
 
         SIGNALRCLIENT_API void send(const std::string& method_name, const signalr::value& arguments = signalr::value(), std::function<void(std::exception_ptr)> callback = [](std::exception_ptr) {}) noexcept;
 
+    protected:
+        friend class hub_connection_builder;
+
+        explicit hub_connection(const std::string& url, trace_level trace_level = trace_level::all,
```

Comment:
Any reason not to just make the ctor private for now? Do we expect users to write classes that derive from hub_connection? In that case they wouldn't be able to use the builder right?